// Starter Proto file, feel free to change name of the file
syntax = "proto3";
package main;
option go_package = "./;kvstore";


/* All RPC service args and rets are supposed to be mentioned here */
/* TODO: Let value just not be string, we can have a oneof field in the proto.*/
message PutKeyArg {
    // Required, Key value to store in kvstore
    string key = 1;
    // Required, Value to store in kvstore.
    string value = 2;
}

message PutKeyRet {
    bool success = 1;
}

message GetKeyArg {
    // Required. Key to fetch the data from KvStore.
    string key = 1;
}

message GetKeyRet {
    bool success = 1;
    string value = 2;
}


/* All RPC services are supposed to be mentioned here */
service KvStoreInterface {
    rpc PutKey(PutKeyArg) returns (PutKeyRet) {}
    rpc GetKey(GetKeyArg) returns (GetKeyRet) {}
}